js/queue.js

```diff
@@ -0,0 +1,37 @@
+var ChatMessagesQueue = function () {
+  this.messageQueue = [];
+};
+
+ChatMessagesQueue.prototype.push = function (message) {
+  this.messageQueue.push(message);
+  return this.messageQueue;
+};
+
+ChatMessagesQueue.prototype.init = function (messages) {
+  this.messageQueue = messages;
+};
+
+ChatMessagesQueue.prototype.getMessages = function () {
+  return this.messageQueue;
+};
+
+ChatMessagesQueue.prototype.sent = function (messages) {
+  var messageQueue = this.messageQueue;
+  messages.forEach(function (sentItem) {
+    messageQueue.forEach(function (message) {
+      if (message.guid === sentItem.guid) {
+        message.sent = true;
```

Comment:
Aren't messages being removed from the queue after being sent instead of being marked as sent?

Reply:
@squallstar no, they are in processing by `chat.message()` function. Which encodes them and save them to the data source. After what sending saved message into this function `onNewMessage()`. Where we remove saved a message from the queue with this method `queue.pull()` and update `Fliplet.Storage` with an updated queue.
That is how it was designed from the start. I just split this order from one large function to smaller ones.